command/connect/envoy/envoy_test.go

```diff
@@ -555,20 +611,24 @@ func TestGenerateConfig(t *testing.T) {
 			// test.
 			srv := httptest.NewServer(testMockAgent(tc.ProxyConfig, tc.GRPCPort))
 			defer srv.Close()
+			client, err := api.NewClient(&api.Config{Address: srv.URL})
+			require.NoError(err)
 
-			// Set the agent HTTP address in ENV to be our mock
-			tc.Env = append(tc.Env, "CONSUL_HTTP_ADDR="+srv.URL)
 			testDirPrefix := testDir + string(filepath.Separator)
 			myEnv := copyAndReplaceAll(tc.Env, "@@TEMPDIR@@", testDirPrefix)
 			defer testSetAndResetEnv(t, myEnv)()
 
 			ui := cli.NewMockUi()
 			c := New(ui)
+			// explicitly set the client to one which can connect to the httptest.Server
+			c.client = client
```

Comment:
Without this change it was impossible to test any settings that enabled TLS. It is possible to start a TLS `httptest` server, but we would still need to inject the TLSConfig into the client.

I think the structure of `Run` could be further improved to make this easier to test, but that would be a much larger change, so I deferred it for now.